dbt/adapters/fabricspark/shortcuts.py

```diff
@@ -153,8 +153,32 @@ def check_exists(self, shortcut: Shortcut):
         # check if the error is ItemNotFound
         if response.status_code == 404:
             return False
-        response.raise_for_status()
+        response.raise_for_status() # raise an exception if there are any other errors
+        # else, check that the target body of the existing shortcut matches the target body of the shortcut they want to create
+        response_json = response.json()
+        response_target = response_json["target"]
+        target_body = shortcut.get_target_body()
+        if response_target != target_body:
+            # if the response target does not match the target body, delete the existing shortcut, then return False so we can create the new shortcut
+            self.delete_shortcut(response_json["path"], response_json["name"])
+            return False
         return True
+    
+    def delete_shortcut(self, shortcut_path: str, shortcut_name: str):
+        """
+        Deletes a shortcut.
+
+        Args:
+            shortcut_path (str): The path where the shortcut is located.
+            shortcut_name (str): The name of the shortcut.
+        """
+        connect_url = f"https://api.fabric.microsoft.com/v1/workspaces/{self.workspace_id}/items/{self.item_id}/shortcuts/{shortcut_path}/{shortcut_name}"
```

Comment:
There's a connect url defined in the class above, can we use that to construct this? Otherwise it looks like that's not being used anywhere.

Reply:
I don't think using that here would be good since the response JSON includes the path and name easily and that is all we need for deletion, but I did refactor `check_exists` to use the `connect_url` function of the shortcut object.